pandas/core/frame.py

```diff
@@ -2639,8 +2642,16 @@ def info(
         buf: Optional[IO[str]] = None,
         max_cols: Optional[int] = None,
         memory_usage: Optional[Union[bool, str]] = None,
+        show_counts: Optional[bool] = None,
         null_counts: Optional[bool] = None,
     ) -> None:
+        if null_counts is not None and show_counts is None:
+            warnings.warn(
+                "null_counts is deprecated. Use show_counts instead",
+                FutureWarning,
+                stacklevel=2,
+            )
```

Comment:
The user will not get this warning when setting both arguments at the same time.
For example, ``df.info(null_counts = True, show_counts=False)``.
This type of invocation is incorrect and should raise an error.

Reply:
Added